ConsoleApp/Program.cs

```diff
@@ -0,0 +1,391 @@
+
+
+
+const string filename = "dump.csv";
+const string filelog = "log.txt";
+(string, DateTime, int, string)[] meetings;
+
+void Menu()
+{
+    Console.Clear();
+    Console.WriteLine("Meeting Booker");
+    Console.WriteLine();
+    Console.WriteLine("1. Create a meeting");
+    Console.WriteLine("2. Show all meetings");
+    Console.WriteLine("3. Update existent meeting");
+    Console.WriteLine("4. Show meeting in room");
+    Console.WriteLine("0. Exit");
+    try
+    {
+        ConsoleKeyInfo key = Console.ReadKey();
+        if (key.Key == ConsoleKey.D0)
+        {
+            Exit();
+        }
+        else if (key.Key == ConsoleKey.D1)
+        {
+            CreateMeeting(null);
+        }
+        else if (key.Key == ConsoleKey.D2)
+        {
+            ShowMeetings();
+        }
+        else if (key.Key == ConsoleKey.D3)
+        {
+            UpdateExistingMeeting(string.Empty);
+        }
+        else if (key.Key == ConsoleKey.D4)
+        {
+            ShowMeetingInRoom();
+        }
+    }
+    catch (Exception e)
+    {
+        LogError(e);
+    }
+}
+
+static void Exit()
+{
+    Environment.Exit(0);
+}
+//метод шукає всі записи по запиту який входе в назву, та дає змогу редагувати запис по індексу
+// string s - строка пошука вертає користувоча на фільтровану сторінку
+void UpdateExistingMeeting(string s)
+{
+    string input = s;
+    Console.Clear();
+    if (string.IsNullOrEmpty(input))
+    {
+        Console.WriteLine("Enter meeting name to update, or press Enter to return:");
+        input = Console.ReadLine();
+    }
+    if (string.IsNullOrEmpty(input)) return;
+    var listmeets = meetings.Where(x => x.Item1.ToLower().Contains(input.ToLower())).ToArray();
+    Console.WriteLine($"Found {listmeets.Length} meetings. Please select meeting to update, or return to Menu");
+    Console.WriteLine("\t0. Return to Menu");
+    int i = 1;
+    foreach (var m in listmeets)
+    {
+        Console.WriteLine($"\t{i}. {m.Item1} {m.Item2} {m.Item3} {m.Item4}");
+        i++;
+    }
+    int num = ReadNumberFromConsole(listmeets.Length);
+    if (num <= 0) return;
+    try
+    {
+        CreateMeeting(num - 1);
+    }
+    catch (Exception e)
+    {
+        LogError(e);
+    }
+    UpdateExistingMeeting(input);
+}
+// Показує всі зустрічи в кімнаті
+void ShowMeetingInRoom()
+{
+    Console.Clear();
+    var m = meetings.Select(x => x.Item4).Distinct().Order().ToArray();
+    Console.WriteLine("Please select number of room, or press Enter to return");
+    int i = 0;
+    foreach (var s in m)
+    {
+        Console.WriteLine($"{i}. {s}");
+        i++;
+    }
+    int num = ReadNumberFromConsole(m.Length - 1);
+    if (num < 0) return;
+    ShowMeetings(m[num]);
+}
+//Нова функція, читає цифри з консолі. maxnum - максимальне число яке можно ввести,0-повернення
+int ReadNumberFromConsole(int maxnum)
+{
+
+    int num = -1;
+    try
+    {
+        while (true)

```

Comment:
what exception can fail in this code? I believe it's safe already without try/catch